orttraining/orttraining/core/graph/gradient_schema_defs.cc

```diff
@@ -1712,6 +1712,38 @@ Return true if all elements are true and false otherwise.
           "Constrain input and output types to float tensors.")
       .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput);
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(BiasGeluGrad_dX)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+      .SetDoc("Computes dX for BiasGeluGrad")
+      .AllowUncheckedAttributes()
+      .Input(0, "dY", "The gradient tensor from output.", "T")
+      .Input(1, "X", "The input tensor. ", "T")
+      .Input(2, "B", "The bias tensor. ", "T")
+      .Output(0, "dX", "Gradient of the input.", "T")
+      .TypeConstraint(
+          "T",
+          {"tensor(float16)", "tensor(float)", "tensor(double)"},
+          "Constrain input and output types to float tensors.")
+      .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput);
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(BiasFastGeluGrad_dX)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+      .SetDoc("Computes dX for FastGeluGrad with bias")
+      .AllowUncheckedAttributes()
+      .Input(0, "dY", "The gradient tensor from output.", "T")
+      .Input(1, "X", "The input tensor. ", "T")
+      .Input(2, "B", "The bias tensor. ", "T")
```

Comment:
>.Input(2, "B", "The bias tensor. ", "T") [](start = 6, length = 40)

Does this need to be optional ? 
since bias in FastGelu is optional ...

Reply:
Nevermind, I see we have another FastGeluGrad for the non-biased case

---
In reply to: [441034996](https://github.com/microsoft/onnxruntime/pull/4146#discussion_r441034996) [](ancestors = 441034996)

Reply:
Right, we assume bias is present in this op.

---
In reply to: [441037129](https://github.com/microsoft/onnxruntime/pull/4146#discussion_r441037129) [](ancestors = 441037129,441034996)